common/transport/mqtt/src/main/java/org/thingsboard/server/transport/mqtt/session/SparkplugNodeSessionHandler.java

```diff
@@ -0,0 +1,288 @@
+import org.thingsboard.server.gen.transport.TransportProtos.SessionInfoProto;
+import org.thingsboard.server.transport.mqtt.MqttTransportContext;
+import org.thingsboard.server.transport.mqtt.MqttTransportHandler;
+import org.thingsboard.server.transport.mqtt.adaptors.JsonMqttAdaptor;
+import org.thingsboard.server.transport.mqtt.adaptors.ProtoMqttAdaptor;
+
+import javax.annotation.Nullable;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+import static org.thingsboard.server.common.transport.service.DefaultTransportService.SESSION_EVENT_MSG_CLOSED;
+import static org.thingsboard.server.common.transport.service.DefaultTransportService.SESSION_EVENT_MSG_OPEN;
+import static org.thingsboard.server.common.transport.service.DefaultTransportService.SUBSCRIBE_TO_ATTRIBUTE_UPDATES_ASYNC_MSG;
+import static org.thingsboard.server.common.transport.service.DefaultTransportService.SUBSCRIBE_TO_RPC_ASYNC_MSG;
+import static org.thingsboard.server.transport.mqtt.util.sparkplug.SparkplugTopicUtil.parseTopic;
+
+/**
+ * Created by nickAS21 on 12.12.22
+ */
+@Slf4j
+public class SparkplugNodeSessionHandler {
+
+    private static final String DEFAULT_DEVICE_TYPE = "default";
+    private static final String CAN_T_PARSE_VALUE = "Can't parse value: ";
+    private static final String DEVICE_PROPERTY = "device";
+
+    private final MqttTransportContext context;
+    private final TransportService transportService;
+    private final TransportDeviceInfo nodeSparkplugInfo;
+    private final UUID sessionId;
+    private final ConcurrentMap<String, Lock> deviceCreationLockMap;
+    private final ConcurrentMap<String, SparkplugSessionCtx> devices = new ConcurrentHashMap<>();
+    private final ConcurrentMap<String, ListenableFuture<SparkplugSessionCtx>> deviceFutures = new ConcurrentHashMap<>();
+    private final ConcurrentMap<MqttTopicMatcher, Integer> mqttQoSMap;
+    private final ChannelHandlerContext channel;
+    private final DeviceSessionCtx deviceSessionCtx;
+    private String nodeTopic;
+
+    public SparkplugNodeSessionHandler(DeviceSessionCtx deviceSessionCtx, UUID sessionId, String nodeTopic) {
+        this.context = deviceSessionCtx.getContext();
+        this.transportService = context.getTransportService();
+        this.deviceSessionCtx = deviceSessionCtx;
+        this.nodeSparkplugInfo = deviceSessionCtx.getDeviceInfo();
+        this.sessionId = sessionId;
+        this.deviceCreationLockMap = createWeakMap();
+        this.mqttQoSMap = deviceSessionCtx.getMqttQoSMap();
+        this.channel = deviceSessionCtx.getChannel();
+        this.nodeTopic = nodeTopic;
+    }
+
+    ConcurrentReferenceHashMap<String, Lock> createWeakMap() {
+        return new ConcurrentReferenceHashMap<>(16, ConcurrentReferenceHashMap.ReferenceType.WEAK);
```

Comment:
Copy-paste from the `GatewaySessionHandler`. Let's merge this now, But the second PR should be refactoring the common code between `GatewaySessionHandler` and `SparkplugNodeSessionHandler` to `AbstractGatewaySessionHandler`